google/cloud/grpc_options_test.cc

```diff
@@ -107,10 +107,47 @@ TEST(GrpcChannelArguments, MakeChannelArguments) {
   expected.SetString("foo", "bar");
   expected.SetString("baz", "quux");
   expected.SetUserAgentPrefix("user_agent");
+  expected.SetInt(GRPC_ARG_KEEPALIVE_TIME_MS, 86400000);
+  expected.SetInt(GRPC_ARG_KEEPALIVE_TIMEOUT_MS, 60000);
 
   CheckGrpcChannelArguments(expected, internal::MakeChannelArguments(opts));
 }
 
+TEST(GrpcChannelArguments, MakeChannelArgumentsDefaults) {
+  grpc::ChannelArguments args;
+
+  // not present
+  EXPECT_FALSE(internal::GetIntChannelArgument(args, GRPC_ARG_KEEPALIVE_TIME_MS));
+
+  // present
+  args.SetInt(GRPC_ARG_KEEPALIVE_TIME_MS, 86400000);
+  auto value = internal::GetIntChannelArgument(args, GRPC_ARG_KEEPALIVE_TIME_MS);
+  ASSERT_TRUE(value.has_value());
+  EXPECT_GT(value, std::chrono::milliseconds(std::chrono::hours(1)).count());
+
+  // not present
+  EXPECT_FALSE(internal::GetIntChannelArgument(args, GRPC_ARG_KEEPALIVE_TIMEOUT_MS));
+
+  // present
+  args.SetInt(GRPC_ARG_KEEPALIVE_TIMEOUT_MS, 60000);
+  auto value_timeout = internal::GetIntChannelArgument(args, GRPC_ARG_KEEPALIVE_TIMEOUT_MS);
+  ASSERT_TRUE(value_timeout.has_value());
+  EXPECT_GT(value_timeout, std::chrono::milliseconds(std::chrono::seconds(1)).count());
+}
+
+TEST(GrpcChannelArguments, MakeChannelArgumentsDefaults_checkOurDefaultsOnly) {
+  auto options = Options{};
```

Comment:
Let's set the channel arguments here (to something other than the defaults) e.g.
```cc
grpc::ChannelArguments args;
args.SetInt(GRPC_ARG_KEEPALIVE_TIME_MS, 1000);
args.SetInt(GRPC_ARG_KEEPALIVE_TIME_MS, 2000);
auto options = Options{}.set<GrpcChannelArgumentsNativeOption>(args);
```

then check that they are the same on the other side of `MakeChannelArguments`.